hydromt/io.py

```diff
@@ -389,11 +389,14 @@ def open_timeseries_from_table(
     """
     kwargs0 = dict(index_col=0, parse_dates=True)
     kwargs0.update(**kwargs)
+    if "date_format" not in kwargs0.keys():
+        kwargs0["date_format"] = "%Y-%m-%d"
```

Comment:
This is a bit risky. What are the inconvenients of letting pandas try to guess format in pd.to_datetime()?

Reply:
Letting Pandas guess at the format generates the following warning: 
```
UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
``` 
Aside from being very bad for performance, this means that you could end up with inconsistent datatype in your column, which is obviously not something you want. In addition this could be a problem for reproducibility, since changes in how pandas makes these guesses can change between minor versions leading to different behavior. Generally I think it is much better to let these things fail explicitly rather than silently because bugs that are caused by this sort of behavior are very difficult to track down. It might be somewhat inconvenient for users, but it is a problem they only have to solve once, and it would protect us from major headache down the line, therefore I think this is worth it.

Reply:
I agree in principle but then I would at least throw an explicit warning with it and try to already correct the warning by updating date_format upstream (data catalog or test data).

Something else which is quite ugly and on our TODO list is that the user currently has no way to pass **kwargs to that function... It is is usually used through io.open_geodataset which only supports so far direct filepath rather than data catalog entries in which you could specify kwargs for pandas and the timeseries table.... so unless we allow the user to pass kwargs I would not implement that change yet.
Not too sure how much work it is though so this might require its own PR.

Reply:
Do we have an issue for this?

Reply:
As you say this might require a bit more than is appropriate for this PR. Especially given that there are some other warnings I've not touched I think it's fine to leave this like this for now. I've reverted this bit back to the original version. I'll make an issue about this